config/oidc_auth.py

```
@@ -0,0 +1,204 @@
+                token,
+                signing_key.key,
+                # Pin the algorithm; never trust the alg in the token header
+                # (defends against alg-confusion / alg=none).
+                algorithms=getattr(settings, "OIDC_ALGORITHMS", ["RS256"]),
+                audience=settings.OIDC_AUDIENCE,
+                issuer=settings.OIDC_ISSUER,
+                options={"require": ["exp", "iat", "iss", "aud", "sub"]},
+                leeway=getattr(settings, "OIDC_LEEWAY", 30),
+            )
+        except jwt.ExpiredSignatureError:
+            raise exceptions.AuthenticationFailed("Token has expired.")
```

Ruff (0.15.18)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)